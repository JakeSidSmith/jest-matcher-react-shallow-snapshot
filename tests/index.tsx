import * as React from 'react';

describe('toMatchShallowSnapshot', () => {
  it('renders some basic markup', () => {
    const element = (
      <div>
        <p>I am a child!</p>
      </div>
    );

    expect(element).toMatchReactShallowSnapshot();
  });

  it('renders a some complex components', () => {
    const handleSubmit = () => null;

    interface MyComponentProps {
      handleSubmit: typeof handleSubmit;
    }

    const Button: React.FunctionComponent = ({ children }) => (
      <button type="submit">{children}</button>
    );

    const ButtonMemo = React.memo<React.PropsWithChildren<{}>>(Button);

    const MyComponent: React.FunctionComponent<MyComponentProps> = props => (
      <>
        <h1>Log in</h1>
        <form onSubmit={props.handleSubmit}>
          <input type="email" name="email" />
          <input type="password" name="password" />
          <ButtonMemo>Log in</ButtonMemo>
        </form>
      </>
    );

    const MyComponentMemo = React.memo(MyComponent);

    expect(
      <MyComponentMemo handleSubmit={handleSubmit} />
    ).toMatchReactShallowSnapshot();
  });
});
