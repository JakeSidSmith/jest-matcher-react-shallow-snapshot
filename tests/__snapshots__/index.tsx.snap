// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchShallowSnapshot renders a some complex components: toMatchShallowSnapshot 1`] = `
<React.Fragment>
  <h1>
    Log in
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      type="email"
    />
    <input
      name="password"
      type="password"
    />
    <React.memo(Button)>
      Log in
    </React.memo(Button)>
  </form>
</React.Fragment>
`;

exports[`toMatchShallowSnapshot renders some basic markup: toMatchShallowSnapshot 1`] = `
<div>
  <p>
    I am a child!
  </p>
</div>
`;
